dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguageHandlerInterface.java

```diff
@@ -324,6 +333,92 @@ default String getURLFromEntry(String dockerEntry, ToolDAO toolDAO) {
         return url;
     }
 
+    // TODO: Implement for DockerHub, then gitlab and seven bridges;
+    default Set<Image> getImagesFromRegistry(String toolsJSONTable) {
+        ArrayList<Map<String, String>> dockerTools = new ArrayList<>();
+        Gson gson = new Gson();
+        dockerTools = (ArrayList<Map<String, String>>)gson.fromJson(toolsJSONTable, dockerTools.getClass());
+
+        Set<String> dockerStrings = new HashSet<>();
+
+        // Eliminate duplicate docker strings
+        for (Map<String, String> tool : dockerTools) {
+            dockerStrings.add(tool.get("docker"));
+        }
+
+        Set<Image> dockerImages = new HashSet<>();
+        for (String image : dockerStrings) {
+            String[] parts = image.split("/");
+
+            Optional<String> response;
+            if (image.startsWith("quay.io/")) {
+                String[] splitDocker = image.split("/");
+                String[] splitTag = splitDocker[2].split(":");
+                if (splitTag.length > 1) {
+                    String repo = splitDocker[1] + "/" + splitTag[0];
+                    String tagName = splitTag[1];
+                    response = getImageResponseFromQuay(image);
+
+                    if (response.isPresent()) {
+                        Map<String, ArrayList<Map<String, String>>> map = new HashMap<>();
+                        Map<String, String> errorMap = new HashMap<>();
+                        map = (Map<String, ArrayList<Map<String, String>>>)gson.fromJson(response.get(), map.getClass());
+                        errorMap = (Map<String, String>)gson.fromJson(response.get(), errorMap.getClass());
+                        if (errorMap.get("error_message") != null) {
```

Comment:
1. Create variable for error_message so you don't repeat it.
2. Do you mean to continue if there's an error? I guess the other code will just fall through harmlessly?

Reply:
No didn't mean to. Changing